### PR TITLE
Publish public version of release instructions

### PR DESCRIPTION
**Pull request by timstclair**
I've gotten a lot of requests to share the release instructions document (publicly), so I scrubbed the google-internal details & converted to markdown. I kept a link to the internal version (which includes some details for Googlers who handle the push).

### CODE REVIEW

File: docs/development/releasing.md [hunk 1]
```diff
@@ -1,3 +1,117 @@
...
+$ sha1sum cadvisor
+$ md5sum cadvisor
+```
+
+Update the github release by uploading the release binary, and filling in the hashes.
+
+## 4. Build & push the Docker images
+
+```
+$ VERSION=v0.23.1  # Example
```
Comment (vishh) [inline docs/development/releasing.md]: Can this step be hidden behind a make rule?

File: docs/development/releasing.md [hunk 2]
```diff
@@ -1,3 +1,117 @@
...
+
+## 3. Build release binary
+
+Command: `make release`
+
+- Make sure your git client is synced to the release cut point
+- Try to build it from the release branch, since we include that in the binary version
+- Verify the ldflags output, in particular check the Version, BuildUser, and GoVersion are expected
+
+Once the build is complete, record the sh1 and md5 hashes:
```
Comment (vishh) [inline docs/development/releasing.md]: Alternatively we can have `make release` output the shas

File: docs/development/releasing.md [hunk 3]
```diff
@@ -1,3 +1,117 @@
...
+
+## 5. Cut the release
+
+Go to https://github.com/google/cadvisor/releases and click "Draft a new release"
+
+- "Tag version" and "Release title" should be preceded by 'v' and then the version. Select the tag pushed in step 2.b
+- Copy an old release as a template (e.g. github.com/google/cadvisor/releases/tag/v0.23.1)
+- Body should start with release notes (from CHANGELOG.md)
+- Next is the Docker image: `google/cadvisor:$VERSION`
+- Next are the binary hashes (from step 3)
```
Comment (vishh) [inline docs/development/releasing.md]: Upload the binary?

Comment (timstclair) [inline docs/development/releasing.md]: Done.

File: docs/development/releasing.md (hunk 1 above)
Comment (timstclair) [inline docs/development/releasing.md]: Done.

**Comment by vishh** [comment]
lgtm

**Pull request merged**